packages/block-editor/src/components/block-patterns-list/index.js

```diff
@@ -77,7 +79,12 @@ function BlockPattern( {
 		>
 			{ ( { draggable, onDragStart, onDragEnd } ) => (
 				<div
-					className="block-editor-block-patterns-list__list-item"
+					className={ clsx(
+						'block-editor-block-patterns-list__list-item',
+						{ 'is-active': isActive } // Apply 'is-active' class if this pattern is active
+					) }
+					role="button" // Add role to make it behave like a button
+					tabIndex={ 0 } // Make the element focusable by keyboard
```

Comment:
The child `Composite.Item` already renders an interactive item with `role="option"` and its "tabbability" is already managed internally by the `Composite` component. Therefore, I don't think this approach is correct in terms of UX, semantics and accessibility

Reply:
Thanks for mentioning this point @ciampo 

I will update this line of codes by removing it since these are already handling in Composite component.

Reply:
I have added these lines since I got  `wp-scripts lint-js` errors while committing the code.

`81:5  error  Visible, non-interactive elements with click handlers must have at least one keyboard listener  jsx-a11y/click-events-have-key-events
  81:5  error  Static HTML elements with event handlers require a role                                         jsx-a11y/no-static-element-interactions`
  
  So shall I keep this?
  
  Please share your thought on this @ciampo 
  
  Thanks.

Reply:
I don't think that this div (the one with class `block-editor-block-patterns-list__list-item`) should be left untouched: no `button` role, no `tabindex`, on `onclick` and `onkeydown` handlers, and no active styles.

There is already a component that is representing the interactive bit of UI, and that is the `Composite.Item` component a few lines below.  We should apply the selected styles to it (similarly to how we already apply focused styles). No need to apply roles or tabindexes, and there is already an `onClick` handler too. The `onkeydown` handler would also be unnecessary, given how the underlying `button` element behaves natively in the browser

Reply:
Done